lib/facter/ipmi.rb

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env ruby
+#
+# IPMI facts, in a format compatible with The Foreman
+#
+# Sending these facts to a puppetmaster equipped with RedHat's Foreman
+# will cause the latter to create the BMC interfaces the next time the
+# Puppet agent runs. One then only has to manually add the access
+# credentials, and presto, you can run basic IPMI actions (e.g. reboot
+# to PXE) from The Foreman's web UI.
+#
+# === Fact Format
+#
+#     ipmi1_ipaddress = 192.168.101.1
+#     ipmi1_subnet_mask = 255.255.255.0
+#     ...
+#
+# where the 1 in "ipmi1" corresponds to the ID of the BMC according to
+# ipmitool lan print.
+#
+# Additionally for compatibility with The Foreman, the first IPMI
+# interface (i.e. the one from ipmi lan print 1) gets all facts
+# repeated as just ipmi_foo:
+#
+#     ipmi_ipaddress = 192.168.101.1
+#     ipmi_subnet_mask = 255.255.255.0
+#     ...
+#
+
+class IPMIChannel
+  public
+  def initialize channel_nr
+    @channel_nr = channel_nr
+    @has_facts = false
+  end
+
+  def load_facts
+    ipmitool_output = `env - $(which ipmitool) lan print #{@channel_nr} 2>&1`
```

Comment:
I've noticed that due to pluginsync, puppet (and mcollective crontab, which loads puppet) throws a warning: `which: no ipmitool in (/bin:/sbin:/usr/bin:/usr/sbin)`. Maybe before executing this line we should make sure that ipmitool is present?
Note: it has no side effect except throwing that warning (and if it's somewhere in a crontab then you get flooded by emails :) ).